### PR TITLE
frontpage: rewrite communities carousel to python and jinja

### PR DESCRIPTION
**Pull request by jennur**
addresses https://github.com/inveniosoftware/invenio-communities/issues/874

See screen recording at https://github.com/inveniosoftware/invenio-communities/pull/966

### CODE REVIEW

File: invenio_app_rdm/theme/views.py [hunk 1]
```diff
@@ -62,11 +64,20 @@ def init_menu():
 #
 # Views
 #
+
+
 def index():
     """Frontpage."""
+    featured_communities = current_communities.service.featured_search(
```
Comment (ntarocco) [inline invenio_app_rdm/theme/views.py:71]: what if an instance will not have feature communities in the home page? This extra search should be avoided in this case.

Comment (zzacharo) [reply invenio_app_rdm/theme/views.py:71]: Plus one on this!

File: invenio_app_rdm/config.py [hunk 2]
```diff
@@ -649,7 +649,7 @@ class UserPreferencesNotificationsSchema(UserPreferencesSchema):
 }
 
 APP_RDM_ROUTES = {
-    "index": "/",
+    "index": ("/", None),
```
Comment (zzacharo) [inline invenio_app_rdm/config.py:652]: by default this should be left as `/` right?

File: invenio_app_rdm/theme/views.py [hunk 3]
```diff
@@ -33,7 +33,19 @@ def create_blueprint(app):
...
+            path, view_func = rule
+
+            if view_func == None:
+                view_func = default_view_func
+
+            return path, name, view_func
+        else:
+            return rule, name, default_view_func
+
+    blueprint.add_url_rule(*create_url_rule(rule=routes["index"], name="index", default_view_func=index))
```
Comment (zzacharo) [inline invenio_app_rdm/theme/views.py:48]: I think it makes sense to use the same way of constructing this for the rest of the routes i.e robots, help_search, help_statistics to be consistent...

File: invenio_app_rdm/theme/views.py [hunk 4]
```diff
@@ -62,6 +74,8 @@ def init_menu():
 #
 # Views
 #
+
```
Comment (zzacharo) [inline invenio_app_rdm/theme/views.py:77]: leftover

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/modules/transition.overrides [hunk 5]
```diff
@@ -2,6 +2,26 @@
     Invenio App RDM Transition Overrides
 ***********************************************/
 
+#carousel-slides {
```
Comment (zzacharo) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/modules/transition.overrides:5]: there is no carousel anymore in invenio-app-rdm so maybe is better to move these in zenodo-rdm? Or are these changes needed for the existing macros?

Comment (jennur) [reply invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/modules/transition.overrides:5]: All the styling is for the macros and/or react component in `invenio-communities`. `zenodo-rdm` is just importing the macro from `invenio-communities`

File: invenio_app_rdm/theme/views.py (hunk 3 above)
Comment (jennur) [reply invenio_app_rdm/theme/views.py:48]: I added the `name` there because I couldn't specify which of the arguments were the `view_func` in `add_url_rule` through the other function (or is there a way?), so it was failing. But I can remove it from the function parameters and just add `None` in the return value?

File: invenio_app_rdm/config.py (hunk 2 above)
Comment (jennur) [reply invenio_app_rdm/config.py:652]: I'm not sure what you mean?

**Review by zzacharo — APPROVED** [review]

File: invenio_app_rdm/theme/views.py [hunk 6]
```diff
@@ -18,6 +18,18 @@
 from invenio_i18n import lazy_gettext as _
 from invenio_users_resources.forms import NotificationsForm
 
+# Generate rule from string or tuple
+def create_url_rule(rule, default_view_func):
+    if isinstance(rule, tuple):
+        path, view_func = rule
+
+        if view_func == None:
```
Comment (zzacharo) [inline invenio_app_rdm/theme/views.py:26]: this can be removed as a tuple should always be used with a view_func

File: invenio_app_rdm/theme/views.py [hunk 7]
```diff
@@ -18,6 +18,18 @@
 from invenio_i18n import lazy_gettext as _
 from invenio_users_resources.forms import NotificationsForm
 
+# Generate rule from string or tuple
+def create_url_rule(rule, default_view_func):
+    if isinstance(rule, tuple):
```
Comment (zzacharo) [inline invenio_app_rdm/theme/views.py:23]: you need a docsting here :)

**Review by zzacharo — APPROVED** [review]
